### ISSUE TITLE
Regression from 2.14.2: dynamic YAML import are ignored (Unresolved imports)

### ISSUE THREAD

**Issue by nicooprat** [webpro/knip#178]
Trying to upgrade from `1.14.0` to latest. I tried every version and it seems that a breaking change was introduced in `2.14.2`: https://github.com/webpro/knip/compare/2.14.1...2.14.2

This used to work:

```ts
// import YAML from /src/i18n/en.yaml
import('@/i18n/en.yaml')
```

With this tsconfig:

```ts
// tsconfig.json
{
  "compilerOptions": {
    "paths": {
      "@/*": [
        "src/*"
      ],
    }
  }
}
```

Now it appears under the `Unresolved imports` section.

I tried to force the path alias through Knip config instead of relying of TypeScript config, with no luck:

```ts
// knip.ts
paths: {
  '@': ['./src'],
},
```

Did I miss something?

Thanks!

**Comment by webpro** [comment]
The things people import in TypeScript :))

The bugfix was basically an improvement for most cases, but yours unfortunately fell the other way. I will add `.yaml` to the list of ignored extensions.

The deeper issue is that `.d.ts` files referenced in `tsconfig.json` are not yet included when TypeScript does its magic for Knip.

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.17.3. See [Release 2.17.3](https://github.com/webpro/knip/releases/tag/2.17.3) for release notes._
